src/main/kotlin/domain/card/Card.kt

```diff
@@ -0,0 +1,68 @@
+package domain.card
+
+import java.util.concurrent.ConcurrentHashMap
+
+class Card private constructor(val denomination: Denomination, val cardType: CardType) {
+
+    val numbers: Set<Int>
+        get() = denomination.numbers
+
+    override fun toString(): String {
+        return "${denomination.text}${cardType.text}"
+    }
+
+    companion object {
+        private val spades = ConcurrentHashMap<Denomination, Card>()
+        private val hearts = ConcurrentHashMap<Denomination, Card>()
+        private val diamonds = ConcurrentHashMap<Denomination, Card>()
+        private val clovers = ConcurrentHashMap<Denomination, Card>()
```

Comment:
종류 구분 없이 가능한 조합의 카드를 모두 생성해둔다음 제공하면 어떨까요?
```suggestion
   private val availableCards: Map<Pair<Denomination, CardType>, Card> = 카드 조합
   
   fun of(denomination: Denomination, cardType: CardType): Card {
       return availableCards[ denomination to cardType ] ?: throw Exception()
   }
```